jena-arq/src/main/java/org/apache/jena/sparql/core/GraphView.java

```diff
@@ -146,7 +148,7 @@ private static Node graphNode(Node gn) {
     public void performAdd( Triple t ) { 
         Node g = graphNode(gn) ;
         if ( Quad.isUnionGraph(g) )
-            throw new GraphViewException("Can't update the default union graph of a dataset") ; 
+            throw new AddDeniedException("Can't update the union graph of a dataset") ; 
```

Comment:
Holy moly, I had no idea we had these. II'm going to remember these, they are way more specific than `UnsupportedOperationException`.

Reply:
It's pushing the "contract". "Denied" suggests to me that it could work in other circumstances but the tests said this was expected so this is what you get.